File: android/src/main/java/com/audiocontext/AudioContextModule.java
package com.audiocontext;

import androidx.annotation.NonNull;

import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.module.annotations.ReactModule;

@ReactModule(name = AudioContextModule.NAME)
public class AudioContextModule extends NativeAudioContextSpec {
  public static final String NAME = "AudioContext";

  public AudioContextModule(ReactApplicationContext reactContext) {
    super(reactContext);
  }

  @Override
  @NonNull
  public String getName() {
    return NAME;
  }

  static {
    System.loadLibrary("react-native-audio-context");
  }

  private static native double nativeMultiply(double a, double b);

  // Example method
  // See https://reactnative.dev/docs/native-modules-android
  @Override
  public double multiply(double a, double b) {
    return nativeMultiply(a, b);
  }
}

File: android/src/main/java/com/audiocontext/nativemodules/AudioContextModule.kt
package com.audiocontext.nativemodules

import com.audiocontext.context.AudioContext
import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.bridge.ReactContextBaseJavaModule
import com.facebook.react.bridge.ReactMethod

class AudioContextModule(private val reactContext: ReactApplicationContext) : ReactContextBaseJavaModule(reactContext) {

  override fun getName(): String {
    return NAME
  }

  @ReactMethod(isBlockingSynchronousMethod = true)
  fun installAudioContext() {
    val audioContext = AudioContext()
    val jsContext = reactContext.javaScriptContextHolder!!.get()
    audioContext.install(jsContext)
  }

  companion object {
    const val NAME: String = "AudioContextModule"
  }
}
